### PR TITLE
[Task][75155] Add logic show trailer screen & logic change theme

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75155: Add logic show trailer screen & logic change theme](https://edu-redmine.sun-asterisk.vn/issues/75155)
## WHAT
- Add logic show trailer screen
- Create UI setting screen
- Add logic change theme
- Add logic show terms and policy
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: MovieHub/MovieHub/SceneDelegate.swift [hunk 1]
```diff
@@ -13,9 +13,26 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
+              let window = window else { return }
+        switch theme {
+        case .dark:
+            window.overrideUserInterfaceStyle = .dark
+        case .light:
+            window.overrideUserInterfaceStyle = .light
+        case .unspecified:
+            window.overrideUserInterfaceStyle = .unspecified
+        }
+    }
```
Comment (ghost) [inline MovieHub/MovieHub/SceneDelegate.swift]:
```suggestion
        window.overrideUserInterfaceStyle = theme
```

File: MovieHub/MovieHub/Scenes/Setting/SettingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            showThemeSelectionAlert { theme in
+                UIApplication.shared.windows.forEach { window in
+                    switch theme {
+                    case .unspecified:
+                        window.overrideUserInterfaceStyle = .unspecified
+                    case .light:
+                        window.overrideUserInterfaceStyle = .light
+                    case .dark:
+                        window.overrideUserInterfaceStyle = .dark
+                    }
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Setting/SettingViewController.swift]: dùng cách như trên

File: MovieHub/MovieHub/Utils/ThemeType.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+//
+//  Created by Duy Nguyễn on 23/04/2024.
+//
+
+import Foundation
+
+enum ThemeType: Int {
+    case unspecified = 0
+    case light = 1
+    case dark = 2
```
Comment (ghost) [inline MovieHub/MovieHub/Utils/ThemeType.swift]: mặc định nó đã là 0 1 2 rồi nên k cần phải thêm e
```suggestion
    case unspecified
    case light
    case dark
```

**Review by ghost — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**